### DISCUSSION TITLE
ImageColumn based on image URL

### DISCUSSION THREAD

**Discussion by aaronmeder** [filamentphp/filament#8806] [Help]
Hey everyone

I would like to show an ImageColumn within my "PostResource" table.

Based on the value of `$post->source` if should show either a local file (like `ImageColumn::make('filename')->disk('media')`) or an image with src set to a dynamic remote image url that I can get with `$post->remoteUrl->full`. 
`$post` would mean the current `Post` Model that is listed on that Filament table row.

The static local file url works fine like this:

```php
public static function table(Table $table): Table
    {
        return $table
            ->columns([
                ImageColumn::make('filename')->disk('media'),
            ])
}
```

However I don't really understand how to put this together with a dynamic remote url based on this Model attribute value.

Thanks for any help.
Aaron

**Comment by avinmaster** [comment]
+1

**Comment by e2d2-dev** [comment]
Hi,

maybe just use the ->getStateUsing() function? 

->getStateUsing(fn($state, $record) => $state ?: $record->remoteUrl->full)
->disk(fn($state) => $state ?: null)
